File: 70-climbing-stairs/climbing-stairs.cpp
#include <cmath>

class Solution {
public:
    int climbStairs(int n) {
        double sqrt5 = std::sqrt(5);
        double phi = (1 + sqrt5) / 2;
        double psi = (1 - sqrt5) / 2;
        return static_cast<int>((std::pow(phi, n + 1) - std::pow(psi, n + 1)) / sqrt5);
    }
};

File: 70-climbing-stairs/climbing-stairs.go
func climbStairs(n int) int {
    sqrt5 := math.Sqrt(5)
    phi := (1 + sqrt5) / 2
    psi := (1 - sqrt5) / 2
    result := (math.Pow(phi, float64(n + 1)) - math.Pow(psi, float64(n + 1))) / sqrt5
    return int(math.Round(result))
}